tools-core/src/main/java/com/hedera/hashgraph/client/core/remote/LargeBinaryFile.java

```diff
@@ -198,6 +186,30 @@ public LargeBinaryFile(final FileDetails fileDetails) throws HederaClientExcepti
 		setShowAdditionalBoxes();
 	}
 
+	@Nullable
+	private JsonObject getJsonObject(final File[] jsons) {
+		final JsonObject details;
+		try {
+			details = readJsonObject(jsons[0].getPath());
+		} catch (final HederaClientException exception) {
+			handleError(exception.getMessage());
+			return null;
+		}
+		return details;
+	}
+
+	private boolean checkInputFile(final File[] jsons, final String type) {
+		assert jsons != null;
+		if (jsons.length != 1) {
+			final var formattedError =
+					String.format("There should be exactly one %s file in the zip archive. We found: %d", type,
+							jsons.length);
+			handleError(formattedError);
+			return true;
```

Comment:
It is counter-intuitive to return `true` as a failed check. Maybe we should rename the method to `failedCheckInputFile`. It would help in the main code to read
```
		if (failedCheckInputFile(jsons, "json")) {
			return;
		}
```